build/general_variables_for_meson.sh

```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env sh
```

Comment:
`/usr/bin/env sh` is pointless cargo culting, `/bin/sh` is *required* by the specs and is the standard guaranteed path (while `/usr/bin/env` is actually not)

Reply:
While ISTR /bin/sh exists on every POSIX system, I also STR that the spec doesn't actually say anything about those existing in /bin, merely that they need to exist in whatever path is available as CS_PATH.

Reply:
I thought about it. In the past I used `env` to be portable but it seems even `env` is not portable. Some systems put it somewhere other than `/usr/bin` and this will break the script. Unfortunately, there is no solution for this in `POSIX`. `command` is an internal command in the shell and does not have a specific path.
Not worth it at all: I use `#!/bin/sh`. 99% `sh` is exist in `/bin`. Indeed, if you know a system that is not like that, I'm sorry, that system is missing something important!

> Applications should note that the standard PATH to the shell cannot be assumed to be either /bin/sh or /usr/bin/sh, and should be determined by interrogation of the PATH returned by [getconf](https://pubs.opengroup.org/onlinepubs/9699919799/utilities/getconf.html) PATH, ensuring that the returned pathname is an absolute pathname and not a shell built-in.
>
>For example, to determine the location of the standard sh utility:
>
>command -v sh
>
>On some implementations this might return:
>
>/usr/xpg4/bin/sh

Ref: https://pubs.opengroup.org/onlinepubs/9699919799/utilities/sh.html#tag_20_117_16

Reply:
Fixed in https://github.com/davmac314/dinit/pull/110/commits/f6fa6b4af12e4102ed74c0a987d760b8bcda8f0c